sympy/core/symbol.py

```diff
@@ -737,3 +739,47 @@ def traverse(symbols, frame):
         del frame  # break cyclic dependencies as stated in inspect docs
 
     return syms
+
+def disambiguate(*iter):
+    """
+    This provides a method to disambiguate symbols or expressions having
+    ambiguate symbols
```

Comment:
Return a Tuple containing the passed expressions with distinct symbols (that appear the same when printed) replaced with numerically subscripted symbols.

Reply:
{I see now that this conflicts with your implementation. But consider when and how you are going to use this. *Please given an example* (for purpose of discussion) of what you would do with an expression that contains distinct but identically printing symbols so you can get a new expression showing them as distinct.

Reply:
Maybe I am thinking of the higher level routine and you are trying to write the lower level routine that it would use; the higher level might be "disambiguate" and the lower leve "_disambiguate_symbols".

Reply:
> Please given an example (for purpose of discussion) of what you would do with an expression that contains distinct but identically printing symbols so you can get a new expression showing them as distinct.
```
>>> x = symbols('x', prime=True)
>>> y = symbols('x', composite=True)
>>> t = x+y; t
x + x      #without disambiguate
>>>  res, = disambiguate(t); res
x + x_1
```

Reply:
> Maybe I am thinking of the higher level routine and you are trying to write the lower level routine that it would use

What should higher level routine do?
At present all the expressions, as well as the symbols, are disambiguated.